charts/dex/templates/rbac.yaml

```diff
@@ -1,6 +1,6 @@
 {{- if .Values.rbac.create }}
 apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
+kind: Role
```

Comment:
As far as I am aware, you can not just change the kind from ClusterRole to Role because this role also grants permissions to create cluster-scoped resources - `customresourcedefinitions`.

1. It would be better to split this role into two roles:
```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: {{ include "dex.fullname" . }}
  labels:
    {{- include "dex.labels" . | nindent 4 }}
rules:
  - apiGroups: ["dex.coreos.com"]
    resources: ["*"]
    verbs: ["*"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: {{ include "dex.fullname" . }}
  labels:
    {{- include "dex.labels" . | nindent 4 }}
rules:
  - apiGroups: ["apiextensions.k8s.io"]
    resources: ["customresourcedefinitions"]
    verbs: ["create"]
```
2. You only need the `{{- if .Values.rbac.create }}` clause for cluster scoped roles and bindings.